Coordinate.cpp

```diff
@@ -14,6 +14,14 @@ NodeCoordinate CellCoordinate::toNode(void) const {
     return NodeCoordinate(x * 2 + 1, y * 2 + 1);
 }
 
+bool NodeCoordinate::isCell() {
+    return !((this->x - 1) % 2 || (this->y - 1) % 2);
```

Comment:
1. `x` and `y` are not ambiguous so `this` is not needed.
2.  !(X || Y) == (X && Y)